test/server/listener_manager_impl_test.cc

```diff
@@ -1424,7 +1424,13 @@ reuse_port: true
 
   auto syscall_result = os_sys_calls_actual_.socket(AF_INET, SOCK_STREAM, 0);
   ASSERT_TRUE(SOCKET_VALID(syscall_result.rc_));
-
+  ON_CALL(os_sys_calls_, bind(_, _, _))
```

Comment:
This looks like it is fixing the issue where `getsockname` on Windows will fail with `WSAEINVAL`: `The socket has not been bound to an address with bind, or ADDR_ANY is specified in bind but connection has not yet occurred.` from: https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-getsockname. Seems like `bind` is faked and causes the error.

Maybe add a comment detailing this and possibly only do this on Windows (though it doesn't hurt on non-Windows, it isn't strictly required). An idea may be to pull this into the `listener_manager_impl_test.h` file in the test setup/constructor guarded by an `#ifdef WIN32` and comment to make it clear why it is an `ON_CALL` and why Windows needs it.

Otherwise, we could reexamine if we can fake all the sys calls so that error becomes irrelevant, I haven't looked deep enough in the implementation to see if that even makes sense.

Reply:
Fair point, I will add a comment to explain why we need this. 

This mock is specific to this test this is why I did not add it to the constructor. No strong opinion on guarding this with an `#idef WIN32`. Personally especially for tests I would avoid the `#ifdef`,